src/pybind/mgr/dashboard/frontend/src/app/shared/services/notification.service.ts

```diff
@@ -123,10 +108,38 @@ export class NotificationService {
           application
         );
       }
+      this.queueToShow(config);
+    }, 10);
+  }
+
+  private queueToShow(config: CdNotificationConfig) {
+    this.cancel(this.queuedTimeoutId);
+    if (!this.queued.find((c) => _.isEqual(c, config))) {
+      this.queued.push(config);
+    }
+    this.queuedTimeoutId = window.setTimeout(() => {
+      this.showQueued();
+    }, 250);
+  }
+
+  private showQueued() {
+    const byTitle = {};
+    let n;
+    while ((n = this.queued.shift())) {
+      if (!byTitle[n.title]) {
+        byTitle[n.title] = [];
+      }
+      byTitle[n.title].push(n);
+    }
+    Object.values(byTitle).forEach((configs: CdNotificationConfig[]) => {
```

Comment:
Just wondering why not using Typescript `for ... in` syntax? I found this syntax is less similar to most languages (vs. `for (key in Map)`):

```suggestion
    for (let n in byTitle) {...
```

Reply:
Nice spotting forgot about that :) I will implement that.

Reply:
Wanted to implement it, but `of` didn't worked here, because `CdNontificationConfig` is not an [iterable object](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Iterators_and_Generators#Iterables).